tests/core/execution/test_configuration_object.py

```diff
@@ -0,0 +1,247 @@
+import pytest
+
+from wasm.datatypes import (
+    Store,
+)
+from wasm.execution import (
+    Configuration,
+    Frame,
+    Label,
+)
+
+
+@pytest.fixture
+def config():
+    return Configuration(Store())
+
+
+def test_configuration_frame_stack_size(config):
+    assert config.frame_stack_size == 0
+
+    frame_a = Frame(None, [], [], 0)
+    config.push_frame(frame_a)
+
+    assert config.frame_stack_size == 1
+
+    frame_b = Frame(None, [], [], 0)
+    config.push_frame(frame_b)
+
+    assert config.frame_stack_size == 2
+
+    assert config.pop_frame() is frame_b
+    assert config.pop_frame() is frame_a
+
+    assert config.frame_stack_size == 0
+
+
+def test_configuration_has_active_frame(config):
+    assert config.has_active_frame is False
+
+    frame_a = Frame(None, [], [], 0)
+    config.push_frame(frame_a)
+
+    assert config.has_active_frame is True
+
+    config.pop_frame()
+
+    assert config.has_active_frame is False
+
+
+def test_configuration_frame_property(config):
+    with pytest.raises(IndexError):
+        config.frame
+
+    frame_a = Frame(None, [], [], 0)
+    config.push_frame(frame_a)
+
+    assert config.frame is frame_a
+
+    frame_b = Frame(None, [], [], 0)
+    config.push_frame(frame_b)
+
+    assert config.frame is frame_b
+
+    assert config.pop_frame() is frame_b
+
+    assert config.frame is frame_a
+
+
+def test_configuration_has_active_label(config):
+    assert config.has_active_label is False
+
+    frame_a = Frame(None, [], [], 0)
+    config.push_frame(frame_a)
+
+    assert config.has_active_label is False
+
+    label_a = Label(0, [], False)
+    config.push_label(label_a)
+
+    assert config.has_active_label is True
+
+    # now push a new frame and there should not be an active label
+    frame_b = Frame(None, [], [], 0)
+    config.push_frame(frame_b)
+
+    assert config.has_active_label is False
+
+    # now pop the frame and should have an active label
+    config.pop_frame()
+
+    assert config.has_active_label is True
+
+    config.pop_label()
+
+    assert config.has_active_label is False
+
+
+def test_configuration_active_label_property(config):
```

Comment:
Just wanted to mention that I appreciate the informative use of whitespace in this module.  I often notice a lack of that simple semantic tool in a lot of people's code.